src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java

```diff
@@ -165,10 +165,10 @@ else if ( id_option.length() != 0){
                             jsonWriter.key("id").value(value);
                         }
                         
-                        GeometryDescriptor defaultGeomType = fType.getGeometryDescriptor();
-                        if(defaultGeomType != null) {
+                        GeometryDescriptor mainGeomType = fType.getGeometryDescriptor();
```

Comment:
Improper rename, we never had a concept of a "main" geometry in GeoTools feature types, but one of a "default" geometry, as made evident by the call to "feature.getDefaultGeometry()" a few lines down. 
Please revert.